src/credentials/providers/credentialsProviderId.ts

```diff
@@ -3,27 +3,28 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { CredentialSourceId } from '../../shared/telemetry/telemetry.gen'
+
 const CREDENTIALS_PROVIDER_ID_SEPARATOR = ':'
 
 export interface CredentialsProviderId {
-    /** Credential type name, e.g. "profile". */
```

Comment:
My understanding is that the user facing text would still be "profile" for credentials coming from shared credentials. The telemetry identifier is internal and not necessarily intended to be surfaced as-is.

@abrooksv ?

Reply:
I was wondering about that too. But then we need a mapping from "telemetryIdentifier to user-facing name". Which currently doesn't exist...

Reply:
We (JetBrains) use the internal ID for storing and referencing, customers only see it in our settings XML files, each credential profile has a displayName method which does the "Profile: $id" generation.

VSC on the other hand exposes its JSON for its launch config as a first class citizen.. this means that the internal ID does will exposed and I don't know if there is a good method.  

The issue I see is that stuff like `ec2:default'` does not really make sense to me since there is only 1 EC2 Instance Metadata Role, and default makes it seem like it should be able to have alternatives. It seems this should just be "EC2" but should allow for a display name that is more understandable than just that IMO....